Source/WebCore/layout/integration/inline/LayoutIntegrationLineLayout.cpp

```diff
@@ -971,6 +971,34 @@ bool LineLayout::hitTest(const HitTestRequest& request, HitTestResult& result, c
     return false;
 }
 
+void LineLayout::shiftLinesBy(LayoutUnit blockShift)
+{
+    if (!m_inlineContent)
+        return;
+    bool isHorizontalWritingMode = WebCore::isHorizontalWritingMode(flow().style().writingMode());
+
+    for (InlineDisplay::Line& line : m_inlineContent->displayContent().lines)
+        line.moveInBlockDirection(blockShift, isHorizontalWritingMode);
+
+    for (InlineDisplay::Box& box : m_inlineContent->displayContent().boxes) {
+        if (isHorizontalWritingMode)
+            box.moveVertically(blockShift);
+        else
+            box.moveHorizontally(blockShift);
+    }
+
+    for (auto& object : m_boxTree.renderers()) {
+        Layout::Box& layoutBox = *object->layoutBox();
+        if (layoutBox.isOutOfFlowPositioned() && layoutBox.style().hasStaticBlockPosition(isHorizontalWritingMode)) {
+            RenderBox& renderer = downcast<RenderBox>(m_boxTree.rendererForLayoutBox(layoutBox));
```

Comment:
```suggestion
            auto& renderer = downcast<RenderBox>(m_boxTree.rendererForLayoutBox(layoutBox));
```

Is this downcast guaranteed to succeed? Are we guaranteed to get a RenderBox here?
Also this might need smart pointers

Reply:
I'm not super familiar with the layout integration layer myself, but I copied the logic from here: https://searchfox.org/wubkat/source/Source/WebCore/layout/integration/inline/LayoutIntegrationLineLayout.cpp#417

```
    for (auto& renderObject : m_boxTree.renderers()) {
        auto& layoutBox = *renderObject->layoutBox();
        if (!layoutBox.isFloatingPositioned() && !layoutBox.isOutOfFlowPositioned())
            continue;
        if (layoutBox.isLineBreakBox())
            continue;
        auto& renderer = downcast<RenderBox>(m_boxTree.rendererForLayoutBox(layoutBox));
```

If that's working off of valid assumptions, then my understanding is that this will work, too.